cleanup/tests/test_daemon.py

```
@@ -0,0 +1,392 @@
+        sock = str(Path(td) / "d.sock")
+        engine = FakeEngine("fake")
+        thread = _serve_in_thread(sock, engine, idle_exit_seconds=3600)
+        try:
+            send_request(sock, "hello", timeout=2.0)
+            assert engine.last_prompt is None
+        finally:
+            _shutdown(sock, thread)
+
+
+def test_cli_warmup_returns_nonzero_when_daemon_missing_no_spawn(tmp_path):
+    r = subprocess.run(
```

Ruff (0.15.12)

[error] `subprocess` call: check for execution of untrusted input

(S603)